runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -127,5 +169,135 @@ private void setDodgyCharges(int dodgyCharges)
 	{
 		this.dodgyCharges = dodgyCharges;
 		config.dodgyNecklace(dodgyCharges);
+		if (currentInfoBoxIDs.get(EquipmentInventorySlot.AMULET) == ItemID.DODGY_NECKLACE)
+		{
+			if (dodgyCharges > 0)
+			{
+				createJewelleryInfobox(ItemID.DODGY_NECKLACE, dodgyCharges, EquipmentInventorySlot.AMULET);
+			}
+			else
+			{
+				removeJewelleryInfobox(EquipmentInventorySlot.AMULET);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
```

Comment:
This is just wrong because it triggers on inventory and bank change

Reply:
If you know of an event that fires off of only the inventory changing, do let me know

Reply:
Check that it equals to the correct item container